src/matlab2tikz.m

```diff
@@ -1945,22 +1938,51 @@ function assertRegularAxes(handle)
     % Also apply the line width if no actual line is there; the markers make use
     % of this, too.
     matlabDefaultLineWidth = 0.5;
-    if m2t.args.semanticLineWidth
-        if ismember(lineWidth, [m2t.args.semanticLineWidthDefinition{:,2}])
-            semStrID = lineWidth == [m2t.args.semanticLineWidthDefinition{:,2}];
-            lineOpts = opts_add(lineOpts, m2t.args.semanticLineWidthDefinition{semStrID,1});
+    if ~isempty(m2t.args.semanticLineWidths)
+        if ismember(lineWidth, [m2t.args.semanticLineWidths{:,2}])
+            semStrID = lineWidth == [m2t.args.semanticLineWidths{:,2}];
+            lineOpts = opts_add(lineOpts, m2t.args.semanticLineWidths{semStrID,1});
         else
-            warning('matlab2tikz:semanticLineWidth',...
+            warning('matlab2tikz:semanticLineWidthNotFound',...
                 ['No semantic correspondance for lineWidth of ''%f'' found.'...
                 'Falling back to explicit export in points.'], lineWidth);
             lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
         end
-    elseif m2t.args.strict ...
-            || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
+    elseif m2t.args.strict || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
         lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
     end
 end
 % ==============================================================================
+function list = configureSemanticLineWidths(m2t)
```

Comment:
What about passing `m2t.args.semanticLineWidths` to this function instead of the monstrous mother object `m2t`? That way, the function is nicely contained.